crates/iota-types/src/scoring_metrics.rs

```diff
@@ -0,0 +1,294 @@
+
+    // Given a VersionedMisbehaviorReport, create a VersionedScoringMetrics struct
+    // with the same values. Used when an authority receives a report from the
+    // network and needs to create a local copy of the metrics contained in it.
+    pub fn from_report(report: &VersionedMisbehaviorReport) -> Self {
+        match report {
+            VersionedMisbehaviorReport::V1(report_v1) => {
+                let committee_size = report_v1.faulty_blocks_provable.len();
+                let metrics = ScoringMetricsV1::new(committee_size);
+                for (i, value) in report_v1.faulty_blocks_provable.iter().enumerate() {
+                    metrics.faulty_blocks_provable[i].store(*value, Ordering::Relaxed);
+                }
+                for (i, value) in report_v1.faulty_blocks_unprovable.iter().enumerate() {
+                    metrics.faulty_blocks_unprovable[i].store(*value, Ordering::Relaxed);
+                }
+                for (i, value) in report_v1.equivocations.iter().enumerate() {
+                    metrics.equivocations[i].store(*value, Ordering::Relaxed);
+                }
+                for (i, value) in report_v1.missing_proposals.iter().enumerate() {
+                    metrics.missing_proposals[i].store(*value, Ordering::Relaxed);
+                }
+                VersionedScoringMetrics::V1(metrics)
+            }
+        }
+    }
+
+    // Given a VersionedScoringMetrics struct, create a VersionedMisbehaviorReport
+    // with the same values. Used when an authority needs to share its local
+    // metrics with the network.
+    pub fn to_report(&self) -> VersionedMisbehaviorReport {
+        match self {
+            VersionedScoringMetrics::V1(metrics) => {
+                let faulty_blocks_provable = metrics
+                    .faulty_blocks_provable
+                    .iter()
+                    .map(|metric| metric.load(Ordering::Relaxed))
+                    .collect();
+                let faulty_blocks_unprovable = metrics
+                    .faulty_blocks_unprovable
+                    .iter()
+                    .map(|metric| metric.load(Ordering::Relaxed))
+                    .collect();
+                let equivocations = metrics
+                    .equivocations
+                    .iter()
+                    .map(|metric| metric.load(Ordering::Relaxed))
+                    .collect();
+                let missing_proposals = metrics
+                    .missing_proposals
+                    .iter()
+                    .map(|metric| metric.load(Ordering::Relaxed))
+                    .collect();
+                VersionedMisbehaviorReport::V1(MisbehaviorReportV1 {
+                    faulty_blocks_provable,
+                    faulty_blocks_unprovable,
+                    equivocations,
+                    missing_proposals,
+                })
+            }
+        }
+    }
+}
+
+pub struct ScoringMetricsV1 {
+    pub faulty_blocks_provable: Vec<AtomicU64>,
```

Comment:
Provide description of each field as inferring the meaning from the name might be confusing and not obvious.

Reply:
no, there is no need for that. I just changed it.